src/box/iproto.cc

```diff
@@ -690,6 +861,16 @@ iproto_connection_try_to_start_destroy(struct iproto_connection *con)
 static inline void
 iproto_connection_close(struct iproto_connection *con)
 {
+	/*
+	 * to_shutdown_list is empty only if con->shutdown_msg is send to
+	 * shutdown_route. to_close_list isn't empty only if con->shutdown_msg
+	 * is send to shutdown_route and iproto_do_close_all_connections didn't
+	 * remove it from iproto_thread->to_close_connections. It means, that
+	 * condition is true only inside shutdown process.
```

Comment:
That's a strange logic. Maybe because I don't understand that lists.
And even if I understood the condition I wouldn't understand why we have to immediately return and don't close. After such a change the function don't have a right to be named 'iproto_connection_close'. It is something else.